tests/unit/local_optimizers/test_continuous_local_optimization.py

```diff
@@ -47,6 +47,47 @@ def test_invalid_algorithm(mocker):
                                     algorithm='Dwayne - The Rock - Johnson')
 
 
+def test_can_set_param_bounds_and_clo_options(mocker):
```

Comment:
should we have a test that ensures the optimization params are actually used when the optimization is called?

Reply:
we discussed that the existing test case that tested this behavior was unclear at first glance. So I refactored the test case to use mocking to be more clear